constructorio-client/src/main/java/io/constructor/client/UserInfo.java

```diff
@@ -58,4 +67,17 @@ private void setClientId(String clientId) throws IllegalArgumentException {
       this.clientId = clientId;
     }
   }
+
+  /**
+   * Validates and sets the user id
+   *
+   * @param userId the user's id
+   */
+  private void setUserId(String userId) throws IllegalArgumentException {
+    if (userId == null || userId.trim().isEmpty()) {
```

Comment:
@jjl014 User id is optional, unlike client id and session id, which are required for user info